MSBLOC.Web/Services/OptionsPrivateKeySource.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GitHubJwt;
+using Microsoft.Extensions.Options;
+using MSBLOC.Web.Models;
+
+namespace MSBLOC.Web.Services
+{
+    public class OptionsPrivateKeySource : IPrivateKeySource
+    {
+        private readonly IOptions<EnvOptions> _optionsAccessor;
+
+        public OptionsPrivateKeySource(IOptions<EnvOptions> optionsAccessor)
+        {
+            _optionsAccessor = optionsAccessor;
+        }
+
+        public TextReader GetPrivateKeyReader()
+        {
+            return new StringReader(CreatePem(_optionsAccessor.Value.GitHubAppPrivateKey));
```

Comment:
I feel like a memory stream would be faster here, since we've got to make a stream anyway.

Reply:
`IPrivateKeySource` is not an interface we control. He is just matching what is there.